src/common/Utility.js

```diff
@@ -420,6 +420,18 @@
                 }
                 return dest;
             }
+        },
+        stringify: function(obj) {
+            var cache = [];
```

Comment:
I think this code assumes there will never be two nodes with the same value (of type 'object'). This doesn't seem like a safe assumption (is it?). Caching the key+value pair would resolve this.